src/storage.cc

```diff
@@ -0,0 +1,219 @@
+// storage.cc
+// Implements the Storage class
+#include "storage.h"
+
+namespace storage {
+
+// - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
+Storage::Storage(cyclus::Context* ctx)
+    : cyclus::Facility(ctx) {
+  cyclus::Warn<cyclus::EXPERIMENTAL_WARNING>("The Storage Facility is experimental.");
+    };
+
+// - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
+// pragmas
+
+#pragma cyclus def schema storage::Storage
+
+#pragma cyclus def annotations storage::Storage
+
+#pragma cyclus def initinv storage::Storage
+
+#pragma cyclus def snapshotinv storage::Storage
+
+#pragma cyclus def infiletodb storage::Storage
+
+#pragma cyclus def snapshot storage::Storage
+
+#pragma cyclus def clone storage::Storage
+
+//- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
+void Storage::InitFrom(Storage* m) {
+
+  #pragma cyclus impl initfromcopy storage::Storage
+  cyclus::toolkit::CommodityProducer::Copy(m);
+}
+
+//- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
+void Storage::InitFrom(cyclus::QueryableBackend* b){
+
+  #pragma cyclus impl initfromdb storage::Storage
+
+  using cyclus::toolkit::Commodity;
+  Commodity commod = Commodity(out_commods_());
+  cyclus::toolkit::CommodityProducer::Add(commod);
+  cyclus::toolkit::CommodityProducer::SetCapacity(commod, throughput);
+}
+
+//- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
+void Storage::EnterNotify() {
+  Facility::EnterNotify();
```

Comment:
Can't hurt - maybe c++ auto-resolves it since storage subclasses cyclus::Facility? Curse you c++!